format_test.go

```diff
@@ -34,14 +39,76 @@ func Test_format(t *testing.T) {
+			grep: map[string]string{
+				"level":    "warning",
+				"function": "GetCurrentState",
+			},
+			expectedOut: `W 18-06-01T06:26:15.261267+0 /apis/core.giantswarm.io/v1alpha1/namespaces/default/kvmclusterconfigs/h0cg9-kvm-cluster-config retrying due to error | operatorkit/controller/resource/retryresource/crud_resource_ops_wrapper.go:71 | event=update | function=GetCurrentState | resource=chartv2 | underlyingResource=chartv2
+	/go/src/github.com/giantswarm/cluster-operator/vendor/github.com/giantswarm/operatorkit/controller/resource/retryresource/crud_resource_ops_wrapper.go:64: 
+	/go/src/github.com/giantswarm/cluster-operator/pkg/v2/resource/chart/current.go:37: 
+	/go/src/github.com/giantswarm/cluster-operator/pkg/v2/resource/chart/resource.go:126: 
+	/go/src/github.com/giantswarm/cluster-operator/vendor/github.com/giantswarm/helmclient/helmclient.go:236: rpc error: code = Unimplemented desc = unknown service grpc.health.v1.Health
+	Tiller installation failed`,
+			errorMatcher: nil,
+		},
+		{
+			name: "case 5: as case 4 but grep for level==warning && function==GetCurrentState && non-existing-key==not_found yielding grepNotFoundError",
+			text: `{"caller":"github.com/giantswarm/cluster-operator/vendor/github.com/giantswarm/operatorkit/controller/resource/retryresource/crud_resource_ops_wrapper.go:71","event":"update","function":"GetCurrentState","level":"warning","message":"retrying due to error","object":"/apis/core.giantswarm.io/v1alpha1/namespaces/default/kvmclusterconfigs/h0cg9-kvm-cluster-config","resource":"chartv2","stack":"[{/go/src/github.com/giantswarm/cluster-operator/vendor/github.com/giantswarm/operatorkit/controller/resource/retryresource/crud_resource_ops_wrapper.go:64: } {/go/src/github.com/giantswarm/cluster-operator/pkg/v2/resource/chart/current.go:37: } {/go/src/github.com/giantswarm/cluster-operator/pkg/v2/resource/chart/resource.go:126: } {/go/src/github.com/giantswarm/cluster-operator/vendor/github.com/giantswarm/helmclient/helmclient.go:236: rpc error: code = Unimplemented desc = unknown service grpc.health.v1.Health} {Tiller installation failed}]","time":"2018-06-01T06:26:15.261267+00:00","underlyingResource":"chartv2"}`,
+			grep: map[string]string{
+				"level":            "warning",
+				"function":         "GetCurrentState",
+				"non-existing-key": "not_found",
+			},
+			expectedOut:  "",
+			errorMatcher: IsGrepNotFound,
 		},
 	}
 
 	disable_colors(true)
 
 	for i, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			out := format([]byte(tc.text))
+			out, err := format([]byte(tc.text), tc.grep)
+
+			switch {
+			case err == nil && tc.errorMatcher == nil:
+				// correct; carry on
+			case err != nil && tc.errorMatcher == nil:
+				t.Fatalf("error == %#v, want nil", err)
+			case err == nil && tc.errorMatcher != nil:
+				t.Fatalf("error == nil, want non-nil")
+			case !tc.errorMatcher(err):
+				t.Fatalf("error == %#v, want matching", err)
+			}
```

Comment:
I think it should return when `tc.errorMatcher != nil`

Reply:
I'm not sure if that makes much difference, but as you suggest it, I'll make it happen. :)

Reply:
I missed those are `Fatalf` not `Errorf`. Sorry for the confusion.